### PR TITLE
feat(gateway): complete Gateway Listener status with route info

### PR DESCRIPTION
**Pull request by michaelbeaumont**
### Summary

This PR adds updating `Gateway` statuses with information about attached routes.

Part of #3617

~Blocked by #3709~

### CODE REVIEW

File: pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go [hunk 1]
```diff
@@ -59,7 +68,70 @@ func gatewayAddresses(gateway *gatewayapi.Gateway, instance *mesh_k8s.GatewayIns
...
 }
 
-func mergeGatewayListenerStatuses(gateway *gatewayapi.Gateway, conditions ListenerConditions) []gatewayapi.ListenerStatus {
+const everyListener = gatewayapi.SectionName("")
+
+type attachedRoutes struct {
+	// num is the number of allowed routes for a listener
+	num int32
+	// invalidRoute can be empty
+	invalidRoute kube_types.NamespacedName
```
Comment (lahabana) [inline pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go:77]: we only ever show the latest version?

Comment (michaelbeaumont) [reply pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go:77]: Hmm no I can include all of them actually.

Comment (michaelbeaumont) [reply pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go]: @lahabana I updated the PR, is that what you meant?

**Review by jakubdyszkiewicz — APPROVED** [review]

File: pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go (hunk 1 above)
Comment (lahabana) [reply pkg/plugins/runtime/k8s/controllers/gatewayapi/gateway_status.go]: Yep thx!

**Pull request merged**